base/libgit2/merge.jl

```diff
@@ -1,5 +1,18 @@
 # This file is a part of Julia. License is MIT: https://julialang.org/license
 
+"""
+    GitAnnotated(repo::GitRepo, commit_id::GitHash)
+    GitAnnotated(repo::GitRepo, ref::GitReference)
+    GitAnnotated(repo::GitRepo, fh::FetchHead)
+    GitAnnotated(repo::GitRepo, comittish::AbstractString)
+
+An annotated git commit carries with it information about how it was looked up and
+why, so that rebase or merge operations have more information about the context of
+the commit. Conflict files contain information about the source/target branches in
+the merge which are conflicting, for instance. An annotated commit can refer to the
+tip of a remote branch, for instance when a [`FetchHead`](@ref) is passed, or to a
+branch head described using `GitReference`.
```

Comment:
Yes. I'm glad the description is useful. It seemed clear to me that these are c-tors but maybe not...